bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -153,13 +145,77 @@ BCStateTran::BCStateTran(
     maxNumOfChunksInVBlock_{calcMaxNumOfChunksInBlock(maxItemSize_, config.maxBlockSize, config.maxChunkSize, true)},
     maxNumOfStoredCheckpoints_{0},
     numberOfReservedPages_{0},
-    randomGen_{randomDevice_()} {
+    randomGen_{randomDevice_()},
+    metrics_component_{concordMetrics::Component("bc_state_transfer",
+                                                  std::make_shared<concordMetrics::Aggregator>())},
+
+    // We must make sure that we actually initialize all these metrics in the
+    // correct order!

```

Comment:
I'd suggest replacing "correct order" with "same order as defined in the header file", to make the intention clearer.

Reply:
Fixed and rebased.